src/lib/gundb/UserStorageClass.js

```diff
@@ -1545,16 +1545,19 @@ export class UserStorage {
    * Calling the server to delete their data
    */
   async deleteAccount(): Promise<boolean> {
+    const zoomId = await this.wallet.getAccountForType('zoomId').replace('0x', '')
+    const zoomSignature = await this.wallet.sign(zoomId, 'zoomId')
+
     let deleteResults = await Promise.all([
-      this.wallet
-        .deleteAccount()
-        .then(r => ({ wallet: 'ok' }))
-        .catch(e => ({ wallet: 'failed' })),
-      API.deleteAccount(this.wallet.getAccountForType('zoomId'))
+      API.deleteAccount(zoomId, zoomSignature)
         .then(r => get(r, 'data.results'))
         .catch(e => ({
           server: 'failed',
         })),
+      this.wallet
```

Comment:
lets delete the account&profile only if server succeeds.
if server fail we should return rejected promise.
delete account popup should change to an error mesage saying delete account failed